Testscripts/Windows/Power-Hibernate.ps1

```diff
@@ -0,0 +1,213 @@
+		# Hence, we are generating constants.sh file again in test script.
+
+		Write-LogInfo "Generating constants.sh ..."
+		$constantsFile = "$LogDir\constants.sh"
+		foreach ($TestParam in $CurrentTestData.TestParameters.param) {
+			Add-Content -Value "$TestParam" -Path $constantsFile
+			Write-LogInfo "$TestParam added to constants.sh"
+		}
+
+		Write-LogInfo "constants.sh created successfully..."
+		#endregion
+
+		#region Add a new swap disk to Azure VM
+		$diskConfig = New-AzDiskConfig -SkuName $storageType -Location $location -CreateOption Empty -DiskSizeGB 1024
+		$dataDisk1 = New-AzDisk -DiskName $dataDiskName -Disk $diskConfig -ResourceGroupName $rgName
+
+		$vm = Get-AzVM -Name $vmName -ResourceGroupName $rgName
+		Start-Sleep -s 30
+		$vm = Add-AzVMDataDisk -VM $vm -Name $dataDiskName -CreateOption Attach -ManagedDiskId $dataDisk1.Id -Lun 1
+		Start-Sleep -s 30
+
+		Update-AzVM -VM $vm -ResourceGroupName $rgName
+		Write-LogInfo "Updated the VM with a new data disk"
+		Write-LogInfo "Waiting for 30 seconds for configuration sync"
+		# Wait for disk sync with Azure host
+		Start-Sleep -s 30
+
+		#region Upload files to master VM
+		foreach ($VMData in $AllVMData) {
+			Copy-RemoteFiles -uploadTo $VMData.PublicIP -port $VMData.SSHPort -files "$constantsFile,$($CurrentTestData.files)" -username $user -password $password -upload
+			Write-LogInfo "Copied the script files to the VM"
+		}
+		#endregion
+
+		# Configuration for the hibernation
+		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "./SetupHbKernel.sh" -RunInBackground -runAsSudo -ignoreLinuxExitCode:$true | Out-Null
+		Write-LogInfo "Executed SetupHbKernel script inside VM"
+
+		# Wait for kernel compilation completion. 60 min timeout
+		$timeout = New-Timespan -Minutes 60
+		$sw = [diagnostics.stopwatch]::StartNew()
+		while ($sw.elapsed -lt $timeout){
+			$vmCount = $AllVMData.Count
+			Wait-Time -seconds 15
+			$state = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password "cat ~/state.txt"
+			if ($state -eq "TestCompleted") {
+				$kernelCompileCompleted = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password "cat ~/constants.sh | grep setup_completed=0"
+				if ($kernelCompileCompleted -ne "setup_completed=0") {
+					Write-LogErr "SetupHbKernel.sh run finished on $($VMData.RoleName) but setup was not successful!"
+				} else {
+					Write-LogInfo "SetupHbKernel.sh finished on $($VMData.RoleName)"
+					$vmCount--
+				}
+				break
+			} elseif ($state -eq "TestSkipped") {
+				Write-LogInfo "SetupHbKernel.sh finished with SKIPPED state!"
+				$testResult = $resultSkipped
+				return $testResult
+			} elseif (($state -eq "TestFailed") -or ($state -eq "TestAborted")) {
+				Write-LogErr "SetupHbKernel.sh didn't finish successfully!"
+				$testResult = $resultAborted
+				return $testResult
```

Comment:
"TestFailed" and "TestAborted" I think they have different status. Return value should distinguish them.